src/gravity.js

```diff
@@ -0,0 +1,508 @@
+  var player = bitsy.player();
+  // player input something, but could not move.
+
+  // force them up if they are doing that
+  if (forceGravityDir) {
+    reallyMovePlayer(player, forceGravityDir)
+    lastMoveMapped = forceGravityDir;
+    return;
+  }
+
+  if (isOnStandableTile(player)) return;
+
+  // otherwise:force them downward if possible.
+  reallyMovePlayer(player, gravityDir);
+  lastMoveMapped = 'down';
+}
+
+window.movePlayerWithGravity = function (dir, axis, amt) {
+  var player = bitsy.player();
+  if (!active) {
+    player[axis] += amt;
+    return;
+  }
+
+  // if player is being pushed, just push em and skip everything else
+  if (forceGravityDir) {
+    reallyMovePlayer(player, forceGravityDir)
+    lastMoveMapped = forceGravityDir;
+    return;
+  }
+
+  // dir is the arrow pressed. realDir is what that arrow translates to in the current gravity inversion...
+  var realDir = mapDir(dir)
+  var horizontal = realDir === 'left' || realDir === 'right';
+
+  var canJetpack = !!jumpPower;
+  // if set to jetpack false, only let them jump if they are standing on something or used jetpack last move
+  if (!jetpack && !wasStandingOnSomething && !wasJetpacking) canJetpack = false
+
+// reset vars and stuff
+  wasJetpacking = false;
+  isClimbing = false;
+
+  // THE BIG MAMA
+  if (realDir === 'up' && (isOnClimbableTile || canJetpack)) {
+
+    if (!isOnClimbableTile) {
+      // if yr moving up, and yr not climbing, then u must be jumping/jetpacking
+      wasJetpacking = true;
+      jumpPower -= 1;
+    } else {
+      isClimbing = true
+    }
+    reallyMovePlayer(player, dir)
+    lastMoveMapped = realDir;
+
+  } else if (horizontal && (wasStandingOnSomething || wasClimbing || isOnClimbableTile || isOnStandableTile(player, dir))){
+
+  // u can always move horizontally if yr standing on smth
+    reallyMovePlayer(player, dir);
+    lastMoveMapped = realDir;
+
+  } else if (horizontal && canMoveHorizontallyWhileFalling()){
+
+    // u can sometimes move horizontally while falling
+    reallyMovePlayer(player, dir);
+    lastMoveMapped = realDir;
+    horizontalFallingMoves += 1;
+
+  } else if (!wasStandingOnSomething) {
+
+    // if none of the other options are valid, and they are falling, then make em fall!
+    reallyMovePlayer(player, gravityDir);
+    lastMoveMapped = 'down';
+
+  } else {
+    // TODO: DIS WHAT HAPPENS WHEN YR STANDING ON THE EDGE!
+  }
+}
+
+// DANGER!
+
+// BRITTLE CODE
+inject(/player\(\)\.x -= 1;/, "movePlayerWithGravity('left', 'x', -1);");
+inject(/player\(\)\.x \+= 1;/, "movePlayerWithGravity('right', 'x', 1);");
+inject(/player\(\)\.y -= 1;/, "movePlayerWithGravity('up', 'y', -1);");
+inject(/player\(\)\.y \+= 1;/, "movePlayerWithGravity('down', 'y', 1);");
+
+// EXTREMELY BRITTLE CODE
+inject(/var ext = getExit\( player\(\)\.room, player\(\)\.x, player\(\)\.y \);/, "else { advanceGravity(); didPlayerMoveThisFrame = true; } var ext = getExit( player().room, player().x, player().y ); ");
```

Comment:
if you capture the original string in the search regex, you can reference it with `$#` to reinsert it in the replacement string
```suggestion
inject(/(var ext = getExit\( player\(\)\.room, player\(\)\.x, player\(\)\.y \);)/, "else { advanceGravity(); didPlayerMoveThisFrame = true; } $1");
```

Reply:
would be helpful to add a comment here describing what this replacement is doing btw; this one's not nearly as self-evident as the movement ones above